qiskit_nature/second_q/mappers/parity_mapper.py

```diff
@@ -54,5 +78,53 @@ def pauli_table(cls, nmodes: int) -> list[tuple[Pauli, Pauli]]:
 
         return pauli_table
 
+    def convert(self, operator: SparsePauliOp) -> SparsePauliOp:
+        """
+        Converts the Operator to tapered one by Z2 symmetries.
+
+        Args:
+            operator: the operator
+        Returns:
+            A new operator whose qubit number is reduced by 2.
+        """
+        num_qubits = operator.num_qubits
+        last_idx = num_qubits - 1
+        mid_idx = num_qubits // 2 - 1
+        sq_list = [mid_idx, last_idx]
+
+        # build symmetries, sq_paulis:
+        symmetries, sq_paulis = [], []
+        for idx in sq_list:
+            pauli_str = ["I"] * num_qubits
+
+            pauli_str[idx] = "Z"
+            z_sym = "".join(pauli_str)[::-1]
+            symmetries.append(z_sym)
+
+            pauli_str[idx] = "X"
+            sq_pauli = "".join(pauli_str)[::-1]
+            sq_paulis.append(sq_pauli)
+
+        symmetries = PauliList(symmetries)
+        sq_paulis = PauliList(sq_paulis)
+        z2_symmetries = Z2Symmetries(symmetries, sq_paulis, sq_list, self._tapering_values)
+
+        return z2_symmetries.taper(operator)
+
     def _map_single(self, second_q_op: FermionicOp) -> PauliSumOp:
-        return ParityMapper.mode_based_mapping(second_q_op, second_q_op.register_length)
+        mapped_op = ParityMapper.mode_based_mapping(
+            second_q_op, second_q_op.register_length
+        ).primitive
+
+        if self.num_particles is not None and mapped_op.num_qubits > 2:
+            reduced_op = self.convert(mapped_op)
+        else:
+            if mapped_op.num_qubits <= 2:
+                logger.warning(
+                    "The original qubit operator only contains %s qubits! Skipping the requested "
+                    "two-qubit reduction!",
+                    mapped_op.num_qubits,
+                )
```

Comment:
I would expect this to only be logged if `self.num_particles is not None`. Currently this would always be logged regardless of whether or not we actually attempted to apply the reduction

Reply:
Fixed